staging/src/k8s.io/apiserver/pkg/admission/plugin/cel/evaluator.go

```diff
@@ -0,0 +1,237 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cel
+
+import (
+	"errors"
+	"fmt"
+	"reflect"
+	"time"
+
+	"github.com/google/cel-go/interpreter"
+
+	admissionv1 "k8s.io/api/admission/v1"
+	authenticationv1 "k8s.io/api/authentication/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apiserver/pkg/admission"
+	"k8s.io/apiserver/pkg/admission/plugin/webhook/generic"
+)
+
+// evaluatorCompiler implement the interface EvaluatorCompiler.
+type evaluatorCompiler struct {
+}
+
+func NewEvaluatorCompiler() EvaluatorCompiler {
+	return &evaluatorCompiler{}
+}
+
+type validationActivation struct {
+	object, oldObject, params, request interface{}
+}
+
+// ResolveName returns a value from the activation by qualified name, or false if the name
+// could not be found.
+func (a *validationActivation) ResolveName(name string) (interface{}, bool) {
+	switch name {
+	case ObjectVarName:
+		return a.object, true
+	case OldObjectVarName:
+		return a.oldObject, true
+	case ParamsVarName:
+		return a.params, true
+	case RequestVarName:
+		return a.request, true
+	default:
+		return nil, false
+	}
+}
+
+// Parent returns the parent of the current activation, may be nil.
+// If non-nil, the parent will be searched during resolve calls.
+func (a *validationActivation) Parent() interpreter.Activation {
+	return nil
+}
+
+// Compile compiles the cel expressions defined in the CELValidations into an Evaluator
+func (c *evaluatorCompiler) Compile(validations []Validation, hasParam bool) Evaluator {
+	if len(validations) == 0 {
+		return nil
+	}
+	compilationResults := make([]CompilationResult, len(validations))
+	for i, validation := range validations {
+		compilationResults[i] = CompileValidatingPolicyExpression(validation.Expression, hasParam)
+	}
+	return NewEvaluator(compilationResults, validations)
+}
+
+// evaluator implements the Evaluator interface
+type evaluator struct {
```

Comment:
So that it doesn't conflict with the Evaluator interface. I went with private struct and the public NewEvaluator() and public interface per comments from @jpbetz above. Tagged you on the conversation.